File: TestClick.Service/Application/Screen/ScreenService.cs
using TestClick.DataAccess;
using TestClick.Interface.Application.Screen;
using TestClick.Model.Application.Screen;
using TestClick.Model.Shared.Param;
using TestClick.Model.Shared.Response;
using Newtonsoft.Json;

namespace TestClick.Service.Application.Screen
{
    public class ScreenService : IScreenService
    {
        private readonly IDataAccessService _db;

        public ScreenService(IDataAccessService db)
        {
            _db = db;
        }

        public async Task<GridResponse<MScreen>?> ScreenSel(SelParamModel<MScreenFilter> param)
        {
            try
            {
                var result = await _db.RetrievalProcedure("Inv.SpScreenSel", JsonConvert.SerializeObject(param));
                return JsonConvert.DeserializeObject<GridResponse<MScreen>>(result);
            }
            catch
            {
                throw;
            }

        }

        public async Task<List<MScreen?>?> ScreenIns(MScreenIns param)
        {
            try
            {
                var result = await _db.ActionProcedure("Inv.SpScreenIns", JsonConvert.SerializeObject(param));
                return JsonConvert.DeserializeObject<List<MScreen?>>(result);
            }
            catch
            {
                throw;
            }
        }
    }
}

File: TestClick.Client/src/app/core/service/screen.service.ts
import { Injectable } from "@angular/core";
import { WebApiService } from "./web-api.service";
import { Observable } from "rxjs";
import { gridResponse, responseModel } from "../../shared/model/response.model";
import { mScreen, mScreenFilter } from "../../shared/model/screen.model";
import { selParamModel } from "../../shared/model/param.model";

@Injectable({
    providedIn: 'root'
})
export class ScreenService {
    baseUrl:string = 'Screen';

    constructor(private api : WebApiService ) {}

    getScreen(param: selParamModel<mScreenFilter>) : Observable<responseModel<gridResponse<mScreen>>> {
       return  this.api.get(`${this.baseUrl}/GetScreen`, param);
    }
}